server: reload of grafana server certs when renewed without restart

chalapat opened the pull request:
**What is this feature?**
This feature is to support reloading the certificates of server when the certificates are renewed, instead of restarting the grafana.

**Why do we need this feature?**

with out this feature, grafana service needs to be restarted upon renewal of the certificates.  In some of the environments the certificates will be renewed daily. To take new certificates into effect, grafana needs to be restarted.

**Who is this feature for?**

For all who needs certificates to be reloaded up on renewal automatically.

**Special notes for your reviewer:**
Connection already established with old certificates needs to refresh the page upon reload. New connections will work with out grafana restart.

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

chalapat committed:
server: reload of grafana server certs when renewed without restart

Signed-off-by: Rao, B V Chalapathi <b_v_chalapathi.rao@nokia.com>

chalapat commented:
Hi @diegommm @mildwonkey @yangkb09 ,
continous-integration/drone/pr stage is in pending state for long time. When i am trying to access, i am getting the following error "Build is blocked, please, contact repo admin in order to proceed".

Can you please help me out what is the issue.

Thanks

DanCech commented:
Thanks for the submission @chalapat !

Being able to pick up a new cert on the fly sounds like a useful feature, but @diegommm and I are concerned about the overhead the current implementation would add to each request. That said, I think this could be implemented in a way that would avoid that overhead.

I suggest creating 2 properties on HTTPServer to hold the certificates and an RWMutex, then run a background goroutine that periodically checks whether the files have changed and updates the certificates (under Lock) if needed. We could potentially have a 3rd property to hold the mtime of the cert file so we can efficiently tell whether they are changed, or we could run readCertificates and compare the return cert list to the current one.

That would decouple checking for a new cert from answering a request, and allow us to only block requests long enough to update the property that holds the cert when we detect that the cert has changed.

It might also be good to add a config setting to control the check interval, and support setting it to 0 to disable the background cert check altogether for people who don't want/need it.

chalapat commented:
Thanks @DanCech  @diegommm . I initially thought of adding such logic, then i have seen prometheus repository and saw there we no such checks put. 
i will add similar such logic

chalapat commented:
@DanCech  @diegommm  I have two way of implementing;
using fsnotify:

- add certfile and keyfile to fsnotify watcher.
- when there is a change to the file an event from fsnotify is received, based on the event update the certs

using periodic watch:

- Based on the configured watchInterval,  check the modified times of the cert and key file.
- if modified time is not same as the  earlier one, read the cert and update the mtimes

Let me know which approach is better?

diegommm commented:
Hi @chalapat! I believe the first approach is great and I personally like it, and we also considered it, but we leaned towards the second one since it avoids introducing a new direct dependency for an optional feature. I suggest using a periodic watch, but add a comment stating that we chose it to avoid importing fsnotify, but if we ever import fsnotify then we can switch the implementation to use it and have a more elegant solution.

Thank you for your contribution!

chalapat committed:
server: reload of grafana server certs when renewed without restart

Signed-off-by: Rao, B V Chalapathi <b_v_chalapathi.rao@nokia.com>

chalapat committed:
server: reload of grafana server certs when renewed without restart

Signed-off-by: Rao, B V Chalapathi <b_v_chalapathi.rao@nokia.com>

chalapat committed:
Merge branch 'main' into reload_cert_support

AgnesToulet committed:
Public Dashboards: Disable email-sharing when there is no license (#80887)

* PublicDashboards: Disable email-shared dashboards when feature is disabled

* fix pubdash creation when it was email-shared

* add feature name const in OSS

* update doc

* Update service.go

* fix test & linter

* fix test

* Update query_test.go

* update tests

* fix imports

* fix doc linter issues

* Update docs/sources/administration/enterprise-licensing/_index.md

* fix after merge

klesh committed:
AuthProxy: Invalidate previous cached item for user when changes are made to any header (#81445)

* fix: sign in using auth_proxy with role a -> b -> a would end up with role b

* Update pkg/services/authn/clients/proxy.go

Co-authored-by: Karl Persson <kalle.persson92@gmail.com>

* Update pkg/services/authn/clients/proxy.go

Co-authored-by: Karl Persson <kalle.persson92@gmail.com>

fabrizio-grafana committed:
Yarn: Clean up PnP fragments (#83138)

konrad147 committed:
Alerting: Fix dashboard nav drawers disappearing (#82890)

Add DashNav modal renderer to handle modals rendered from Toolbar buttons

fabrizio-grafana committed:
Run analysis steps only on `grafana/grafana` (#83185)

fabrizio-grafana committed:
Allow build and release process for Zipkin plugin (#83116)

ppcano committed:
Alerting docs: fix broken link and apply the Writer toolkit guideline to other links (#83209)

* Fix `/docs/reference` cannot be used within admonitions

* Remove `relref` links to apply Writers toolkit guidelines

ashharrison90 committed:
Navigation: Scroll the active menu item into view when menu is an overlay (#83212)

scroll the active menu item into view even when not docked

joshhunt committed:
MigrateToCloud: Add API interface for frontend (#83215)

Add RTK Query mocks

jackw committed:
Monaco Editor: Load via ESM (#78261)

* chore(monaco): bump monaco-editor to latest version

* feat(codeeditor): use esm to load monaco editor

* revert(monaco): put back previous version

* feat(monaco): setup MonacoEnvironment when bootstrapping app

* feat(monaco): load monaco languages from registry as workers

* feat(webpack): clean up warnings, remove need to copy monaco into lib

* fix(plugins): wip - remove amd loader workaround in systemjs hooks

* chore(azure): clean up so QueryField passes typecheck

* test(jest): update config to fix failing tests due to missing monaco-editor

* test(jest): update config to work with monaco-editor and kusto

* test(jest): prevent message eventlistener in nodeGraph/layout.worker tripping up monaco tests

* test(plugins): wip - remove amd related tests from systemjs hooks

* test(alerting): prefer clearAllMocks to prevent monaco editor failing due to missing matchMedia

* test(parca): fix failing test due to undefined backendSrv

* chore: move monacoEnv to app/core

* test: increase testing-lib timeout to 2secs, fix parca test to assert dom element

* feat(plugins): share kusto via systemjs

* test(e2e): increase timeout for checking monaco editor in exemplars spec

* test(e2e): assert monaco has loaded by checking the spinner is gone and window.monaco exists

* test(e2e): check for monaco editor textarea

* test(e2e): check monaco editor is loaded before assertions

* test(e2e): add waitForMonacoToLoad util to reduce duplication

* test(e2e): fix failing mysql spec

* chore(jest): add comment to setupTests explaining need to incresae default timeout

* chore(nodegraph): improve comment in layout.worker.utils to better explain the need for file

leeoniya committed:
StateTimeline: Properly type tooltip prop, remove ts-ignore (#83189)

mdcruz committed:
Docs: update grafana fundamentals (#83075)

* docs: update grafana fundamentals

* fix: lint issues

* docs: added more content on grafana fundamentals tutorial

gabor committed:
mysql: do not use unexported grafana-core config (#83062)

* mysql: do not use unexported grafana-core config

* updated test

soniaAguilarPeiron committed:
Alerting: Fix saving evaluation group. (#83188)

fix saving evaluation group

joey-grafana committed:
Tempo: Remove trace to metrics feature toggle (#82884)

* Remove trace to metrics feature toggle

* Fix after merge

jackw committed:
Chore: Align usage of tsconfig in yarn workspaces to 1.3.0-rc1 (#83160)

chore(tsconfig): align all usage in workspaces to 1.3.0-rc1

imatwawana committed:
Docs: update import troubleshoot dashboards links (#83124)

* Updated links to former manage dashboards content

* Removed links to manage dashboards and added export content to Sharing page

* Replaced grafana links with cloud docs links

* Removed trailing slash from link

* trigger CI

---------

Co-authored-by: Jack Baldry <jack.baldry@grafana.com>

jackw committed:
Chore: Bump Lerna 8.x.x (#83233)

* chore(lerna): bump lerna to 8.x.x

* chore(lerna): run lerna repair command to update lerna.json

* ci(drone): use raw output (no quotes) when updating package.json version

* ci(drone): update config file

gabor committed:
postgres: do not use unexported grafana-core config (#83241)

* postgres: do not use unexported grafana-core config

* fixed wrong value

renovate[bot] committed:
Update dependency rc-drawer to v7 (#83162)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
Update dependency postcss-loader to v8 (#83161)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

grobinson-grafana committed:
Alerting: Basic support for time_intervals (#83216)

This commit adds basic support for time_intervals, as mute_time_intervals
is deprecated in Alertmanager and scheduled to be removed before 1.0.
It does not add support for time_intervals in API or file provisioning,
nor does it support exporting time intervals. This will be added in
later commits to keep the changes as simple as possible.

renovate[bot] committed:
Update dependency sass-loader to v14 (#83255)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

mgyongyosi committed:
Chore: Allow self-serve for ssoSettingsApi feature toggle (#83140)

* Set AllowSelfServe to true ssoSettingsApi

* Align ft registry validation with latest changes

grobinson-grafana committed:
Alerting: Keep order of time and mute time intervals consistent (#83257)

harisrozajac committed:
Dashboard-Scene: View panel as table in edit mode (#83077)

* WIP: working functionality

* betterer

* Fully working: Alerts show up, toggling table view doesn't update viz type in options pane

* betterer

* improve

* betterer

* Refactoring a bit

* wrong step

* move data provider to vizPanel

* Update

* update

* More refactorings

* Fix InspectJsonTab tests (except 1); remove obsolete PanelControls

* Fixed test

* Update

* minor fix

---------

Co-authored-by: Torkel Ödegaard <torkel@grafana.com>

aangelisc committed:
Docker: Add workaround for building on `arm64` (#83242)

Add workaround for building on arm64

torkelo committed:
DataTrails: Remove the adhoc filters label (#83237)

evictorero committed:
Scenes: Render old snapshots (#82277)

knylander-grafana committed:
[DOC] Tempo data source: fix broken link and clarify traces to profile (#83135)

* Update Tempo data source to fix broken link

* Correct profiles content

* Move explanation table up

* Chagnes from prettier

* Resolve conflict

ahuarte47 committed:
Table Component: Improve text-wrapping behavior of cells (#82872)

* Fix text-wrapping of cells in Tables

* Set wordbreak on hover for long texts without spaces

codeincarnate committed:
Table: Fix units showing in footer after reductions without units (#82081)

* Add preservesUnits property to reducer registry items

* Hide units when appropriate

* Prettier

* some code cleanup

* Prevent error when no stat is selected.

---------

Co-authored-by: nmarrs <nathanielmarrs@gmail.com>

chalapat commented:
@diegommm @DanCech  can you please enable the **continuous-integration/drone/pr** stage

konrad147 committed:
Alerting: Add Loki ASH tests checking proper rendering of timeline chart component (#83235)

Add Loki ASH tests checking proper rendering of timeline chart component

renovate[bot] committed:
Update dependency stylelint-config-sass-guidelines to v11 (#83253)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
Update dependency webpack-dev-server to v5 (#83258)

* Update dependency webpack-dev-server to v5

* update webpack.hot config (is this even used anymore?)

---------

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>
Co-authored-by: Ashley Harrison <ashley.harrison@grafana.com>

renovate[bot] committed:
Update dependency stylelint to v16 (#83252)

* Update dependency stylelint to v16

* remove stylelint-config-prettier since it's no longer necessary

---------

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>
Co-authored-by: Ashley Harrison <ashley.harrison@grafana.com>

Umaaz committed:
fix(build): make python and gcc are needed for some yarn dependencies… (#83228)

fix(build): make python and gcc are needed for some yarn dependencies to build

ashharrison90 committed:
Chore: replace `react-popper` with `floating-ui` in time pickers (#82640)

* replace react-popper with floating-ui in RelativeTimeRangePicker

* replace react-popper with floating-ui in DateTimePicker

ashharrison90 committed:
E2C: Add initial empty state (#83232)

* update subtitle

* add empty state

* rename to InfoPaneLeft/Right

* use &apos;

* add "direction" prop to Box

* update subtitle

* Revert "add "direction" prop to Box"

This reverts commit 99f82a27c732541fe9ca0f1dc87e6db0f6eb72fc.

grobinson-grafana committed:
Alerting: Update swagger specs (#83260)

ashharrison90 committed:
Box: add `direction` prop to `Box` (#83296)

add "direction" prop to Box

sunker committed:
E2E: Add plugin-e2e scenario verification tests (#79969)

* add playwright test and plugin-e2e

* run tests in ci

* add ds config tests

* add panel edit tests

* add annotation test

* add variable edit page tests

* add explore page tests

* add panel plugin tests

* add readme

* remove comments

* fix broken test

* remove user.json

* remove newline in starlark

* fix lint issue

* ignore failure of playwright tests

* update code owners

* add detailed error messages in every expect

* update message frame

* fix link

* upload report to gcp

* echo url

* add playwright developer guide

* bump plugin-e2e

* add custom provisioning dir

* update plugin-e2e

* remove not used imports

* fix typo

* minor fixes

* use latest version of plugin-e2e

* fix broken link

* use latest plugin-e2e

* add feature toggle scenario verification tests

* bump version

* use auth file from package

* fix type error

* add panel data assertions

* rename parent dir and bump version

* fix codeowners

* reset files

* remove not used file

* update plugin-e2e

* separate tests per role

* pass prov dir

* skip using provisioning fixture

* wip

* fix permission test

* move to e2e dir

* fix path to readme

* post comment with report url

* format starlark

* post comment with report url

* post comment with report url

* fix token

* make test fail

* fix exit code

* bump version

* bump to latest plugin-e2e

* revert reporting message

* remove comments

* readding report comment

* change exit code

* format starlark

* force test to fail

* add new step that posts comment

* fix link

* use latest playwright image

* fix failing test

* format starlark

* remove unused fixture

Co-authored-by: Marcus Andersson <marcus.andersson@grafana.com>

---------

Co-authored-by: Marcus Andersson <marcus.andersson@grafana.com>

aangelisc committed:
CI: Remove inline-builds flag (#83306)

Remove inline-builds

raymalt committed:
Unit: Add SI prefix for empty unit (#79897)

* Unit: Add SI prefix for empty unit

* Units: Change name from SI prefix to SI short

toddtreece committed:
Chore: Add go workspace (#83191)


---------

Co-authored-by: ismail simsek <ismailsimsek09@gmail.com>

joey-grafana committed:
Tempo: Add template variable interpolation for filters (#83213)

* Interpolate template variables in filters

* Add tests

ashharrison90 committed:
Grid: Add `alignItems` prop (#83314)

add alignItems props to Grid

darrenjaneczek committed:
datatrails: detect if current trail state is bookmarked (#83283)

* fix: detect if current trail state is bookmarked

ashharrison90 committed:
Chore: replace `react-popper` with `floating-ui` in `Popover` (#82922)

* replace react-popper with floating-ui in Popover

* update HoverCard

* fix unit tests

* mock useTransitionStyles to ensure consistent unit test results

torkelo committed:
DataTrails: Sticky controls (#83286)

* DataTrails: Sticky controls

* Update

gelicia committed:
Explore: Use rich history local storage for autocomplete (#81386)

* move autocomplete logic

* Tests

* Write to correct history

* Remove historyUpdatedAction and related code

* add helpful comments

* Add option to mute all errors/warnings for autocomplete

* Add back in legacy local storage query history for transition period

* Move params to an object for easier use and defaults

* Do not make time filter required

* fix tests

* change deprecation version and add issue number

Jguer committed:
IAM: Remove fully rolled out feature toggles (#83308)

* remove anon stat ft

* remove split scope flag

* remove feature toggle from frontend

IevaVasiljeva committed:
RBAC: add kind, attribute and identifier to annotation permissions during the migration (#83299)

add kind, attribute and identifier to annotation permissions during the migration

IevaVasiljeva committed:
IP range AC for data sources: compare the base of the URL only (#83305)

* compare the base of the URL and ignore the path

* change the logic to compare scheme and host explicitly

* fix the test

aangelisc committed:
CI: Add retry for yarn install (#83317)

Add retry for yarn install

chri2547 commented on docs/sources/setup-grafana/configure-grafana/_index.md:
```diff
@@ -286,6 +286,17 @@ Path to the certificate file (if `protocol` is set to `https` or `h2`).
+### certs_watch_interval
+
+Periodicity at which `cert_key` and `cert_file` would be watched for changes.
+By default it is disabled. If enabled, then periodically `cert_key` and `cert_file`
```
```suggestion
Disabled, by default. When enabled, `cert_key` and `cert_file`
```

chri2547 commented on docs/sources/setup-grafana/configure-grafana/_index.md:
```diff
@@ -286,6 +286,17 @@ Path to the certificate file (if `protocol` is set to `https` or `h2`).
+
+Periodicity at which `cert_key` and `cert_file` would be watched for changes.
+By default it is disabled. If enabled, then periodically `cert_key` and `cert_file`
+are watched for changes, if there is change seen then those are certs are loaded automatically.
```
```suggestion
are periodically watched for changes. If there is a change, then those are certs are loaded automatically.
```

chri2547 commented on docs/sources/setup-grafana/configure-grafana/_index.md:
```diff
@@ -286,6 +286,17 @@ Path to the certificate file (if `protocol` is set to `https` or `h2`).
+are watched for changes, if there is change seen then those are certs are loaded automatically.
+
+{{% admonition type="warning" %}}
+After the new certificates are loaded, connections which are established with old certificates
```
```suggestion
After the new certificates are loaded, connections with old certificates
```

chri2547 reviewed (commented):
@chalapat , please take a look at my wording suggestions to ensure I haven't changed the technical meaning you intend to communicate. Let me know if I'm off base and can work with you on finalizing. Thank you for the contribution!

brendamuir committed:
Alerting docs: rework create alert rules definition and topic (#83220)

* Alerting docs: rework create alert rules definition and topic

* ran prettier

* corrects note

* adds configure notifications section

* update branch

* corrects links

* parts of alert rule creation

* Update docs/sources/alerting/configure-notifications/_index.md

Co-authored-by: Pepe Cano <825430+ppcano@users.noreply.github.com>

* moved section

* updates recording rule steps

* gets rid of configure integrations topic

* deletes configure integrations topic

* deletes links

* ran prettier

* Include contact point links

* phase 2 sorting out manage notifications

* manage notification changes

* manage notification updates

* finishing touches

* fixes links

* link fixes

* more link fix

* link fixes

* ran prettier

---------

Signed-off-by: Jack Baldry <jack.baldry@grafana.com>
Co-authored-by: Pepe Cano <825430+ppcano@users.noreply.github.com>
Co-authored-by: Jack Baldry <jack.baldry@grafana.com>

gelicia committed:
remove oss from security config docs (#82936)

baldm0mma committed:
Table: Add ability for Table to render Standard Options "No value" value when DataFrames or field values are empty (#82948)

* baldm0mma/no_value_message/ add fieldConfig to Table props

* baldm0mma/no_value_message/ add noValuesDisplayText to table props

* baldm0mma/no_value_message/ add fieldConfig to tablePanel

* baldm0mma/no_value_message/ add tests

* baldm0mma/no_value_message/ update test values

* baldm0mma/no_value_message/ update args in tests

* baldm0mma/no_value_message/ update with NO_DATA_TEXT const

darrenjaneczek committed:
style: datatrails: use Tag on cards (#83198)

* style: datatrails: use Tag on cards

darrenjaneczek committed:
style: datatrails: indicate prometheus data source (#83199)

* style: indicate prometheus data source

jdbaldry committed:
Revert "Alerting docs: rework create alert rules definition and topic" (#83328)

toddtreece committed:
Chore: Update wire to v0.6.0 using bingo (#83323)

undef1nd committed:
Chore: Remove docs and kinds report generators (#83277)

* Chore: Remove codegen for docs

* Remove kindsysreport

toddtreece committed:
Chore: Update drone signature (#83334)

toddtreece committed:
Chore: Add env var check to `make drone` (#83337)

toddtreece committed:
K8s: Add apimachinery and apiserver packages (#83190)

imatwawana committed:
Docs: restructure Configure field overrides (#81833)

* Removed view and delete overrides sections

* Added examples heading and moved examples down one heading level

* Added override rules section and removed rule definitions from task

* Added supported visualizations section and table and docs ref links

* Docs: edit Configure field overrides (#81834)

* Formatted note

* Added missing content and general edits

* Updated screenshots and examples and general edits

* Fix small formatting issues

* Fixed links

* Uploaded images to admin, updated image links, and removed local images

* Swapped figure shortcode for simple Markdown

itsmylife committed:
Chore: Add a key prop to warning component to prevent "should have a unique key prop" error (#83340)

add a key prop to prevent "should have a unique key prop" warning/error on the browser console

harisrozajac committed:
Dashboard-Scene: Show empty state after removing last panel (#83114)

* Show empty state after removing last panel

* betterer

* Refactor isEmpty state update in DashboardScene.tsx

* don't need viewPanelScene check

* track isEmpty through a behavior

* Fix test

* Add test for empty state

* minor fix

* Refactor isEmpty check

* Don't use const

* clean up

leeoniya committed:
VizTooltips: Use global portal (#81986)

Co-authored-by: Adela Almasan <adela.almasan@grafana.com>

toddtreece committed:
K8s: Update codegen to support new packages (#83347)

---------

Co-authored-by: Charandas Batra <charandas.batra@grafana.com>

darrenjaneczek committed:
fix: datatrails: start step is "metric select" after reinitializing from URL (#83179)

fix: datatrails: start step is "metric select"

If restoring from a saved data trail, or URL,
the selection of metric becomes a second step.
The first step is always without a metric, and
displays the metric selection scene.

sunker committed:
plugin-e2e: Fix flaky test (#83370)

fix mock issue

joey-grafana committed:
Tempo: Add support for ad-hoc filters (#83290)

* Add support for ad-hoc filters

* Add tests

* Update types

spinillos committed:
Kinds: Generate k8 resources without use kindys/thema (#83310)

Generate k8 resources reading cue file directly instead of use thema/kindsys binding

ashharrison90 committed:
E2C: Add cloud landing page (#83316)

* restructure and create cloud empty state

* use new Grid prop and run i18n:extract

renovate[bot] committed:
Update dependency @grafana/aws-sdk to v0.3.2 (#83374)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

zserge committed:
Chore: Add support bundle for folders (#83360)

* add support bundle for folders

* fix ProvideService in tests

* add a test for collector

renovate[bot] committed:
Update dependency @grafana/faro-core to v1.3.9

gamab committed:
AuthN: Remove embedded oauth server (#83146)

* AuthN: Remove embedded oauth server

* Restore main

* go mod tidy

* Fix problem

* Remove permission intersection

* Fix test and lint

* Fix TestData test

* Revert to origin/main

* Update go.mod

* Update go.mod

* Update go.sum

fabrizio-grafana committed:
Tempo: Remove duplicated code (#81476)

eledobleefe committed:
ReturnToPrevious: Add guidelines to `returnToPrevious` hook (#83219)

renovate[bot] committed:
Update dependency @grafana/faro-web-sdk to v1.3.9

renovate[bot] committed:
Update dependency @grafana/faro-web-sdk to v1.3.9

andresmgot committed:
Datasource API: Add config to ctx (#83386)

renovate[bot] committed:
Update dependency @react-types/shared to v3.22.1

Commit:
Image Rendering: Add settings for default width, height and scale (#82040)

* Add Image width & height

* ability to change default width, height and scale

* default ini

* Update conf/defaults.ini

Co-authored-by: Agnès Toulet <35176601+AgnesToulet@users.noreply.github.com>

* Update pkg/setting/setting.go

Co-authored-by: Agnès Toulet <35176601+AgnesToulet@users.noreply.github.com>

* Update pkg/setting/setting.go

Co-authored-by: Agnès Toulet <35176601+AgnesToulet@users.noreply.github.com>

* Added docs, changed frontend

* Update conf/defaults.ini

Co-authored-by: Agnès Toulet <35176601+AgnesToulet@users.noreply.github.com>

* Update conf/defaults.ini

Co-authored-by: Agnès Toulet <35176601+AgnesToulet@users.noreply.github.com>

* Update conf/defaults.ini

Co-authored-by: Agnès Toulet <35176601+AgnesToulet@users.noreply.github.com>

* Update docs/sources/setup-grafana/configure-grafana/_index.md

Co-authored-by: Agnès Toulet <35176601+AgnesToulet@users.noreply.github.com>

* Update pkg/api/dtos/frontend_settings.go

Co-authored-by: Agnès Toulet <35176601+AgnesToulet@users.noreply.github.com>

* Update pkg/api/frontendsettings.go

Co-authored-by: Agnès Toulet <35176601+AgnesToulet@users.noreply.github.com>

* Update pkg/api/render.go

Co-authored-by: Agnès Toulet <35176601+AgnesToulet@users.noreply.github.com>

* add query float 64

* Update packages/grafana-runtime/src/config.ts

Co-authored-by: Agnès Toulet <35176601+AgnesToulet@users.noreply.github.com>

* Update public/app/features/dashboard/components/ShareModal/utils.ts

Co-authored-by: Agnès Toulet <35176601+AgnesToulet@users.noreply.github.com>

* spacing

* fix tests

* Update docs/sources/setup-grafana/configure-grafana/_index.md

Co-authored-by: Christopher Moyer <35463610+chri2547@users.noreply.github.com>

* Update docs/sources/setup-grafana/configure-grafana/_index.md

Co-authored-by: Christopher Moyer <35463610+chri2547@users.noreply.github.com>

* Update docs/sources/setup-grafana/configure-grafana/_index.md

Co-authored-by: Christopher Moyer <35463610+chri2547@users.noreply.github.com>

---------

Co-authored-by: Agnès Toulet <35176601+AgnesToulet@users.noreply.github.com>
Co-authored-by: Christopher Moyer <35463610+chri2547@users.noreply.github.com>

renovate[bot] committed:
Update dependency @swc/core to v1.4.2

mdvictor committed:
Scenes: Add new vizualisation functionality (#83145)

* wip

* tests + refactor ad panel func

* PR mods

idafurjes committed:
Restructure cloudmigration service (#83211)

* Restructure cloudmigation service

* Adjust codewoners and wire

* Comment out unused metrics

hugohaggmark committed:
Docs: Update HALL_OF_FAME.md (#79775)

Co-authored-by: Jack Baldry <jack.baldry@grafana.com>

gabor committed:
update go.mod/sum/work (#83402)

* update go.mod/sum/work

* updated codeowners

eleijonmarck committed:
Team LBAC: Add doc description of lbac rule (#83384)

* chore: add doc description of lbac rule

* spellling

* Update docs/sources/administration/data-source-management/teamlbac/_index.md

Co-authored-by: Alexander Zobnin <alexanderzobnin@gmail.com>

---------

Co-authored-by: Alexander Zobnin <alexanderzobnin@gmail.com>

mgyongyosi committed:
Auth: OAuth strategy load extra fields separately (#83408)

Load extra fields separately

iwysiu committed:
CloudWatch: Remove unused handleGetRegions and regionCache (#83333)

* remove unused handleGetRegions and regionCache

* cleanup

ashharrison90 committed:
E2C: Initial config to switch between cloud and onprem pages (#83380)

hacky config to switch between cloud and onprem pages

darrenjaneczek committed:
datatrails: fix: indicate `__ignore_usage__` for preview queries (#82291)

* fix: indicate __ignore_usage__ for preview queries

* fix: further remove var that shouldn't be there

* refactor: move previewPanel code and add tests

* fix: use simpler `AdHocFiltersVariable` state

Co-authored-by: Andre Pereira <adrapereira@gmail.com>

---------

Co-authored-by: Andre Pereira <adrapereira@gmail.com>

tskarhed committed:
Docs: Update help keyboard shortcut (#83397)

renovate[bot] committed:
Update dependency @types/chance to v1.1.6 (#83401)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
Update dependency @types/common-tags to v1.8.4 (#83410)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

ryantxu committed:
Live: Improve the debug panel and add a devenv dashbaord (#83350)

renovate[bot] committed:
Update dependency @types/d3-force to v3.0.9 (#83413)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

dmihai committed:
SSO: update info field from SocialBase struct (#83407)

update info field from SocialBase struct

renovate[bot] committed:
Update dependency @types/js-yaml to v4.0.9 (#83417)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

itsmylife committed:
InfluxDB: Fix fetching tag values only for selected measurement (#83353)

show tag values only for selected measurement

renovate[bot] committed:
Update dependency @types/lucene to v2.1.7 (#83422)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

usmangt committed:
Docs/grafana helm (#80390)

* added the helm project

* added page metadata

* added the intro section

* fixed menuTitle

* added section i.e. Setting up the Grafana Helm repository

* added the deployment section

* finished the deploying grafana section

* completed access grafana section

* updating changes

* added persistent storage section

* added debugging section

* fixed typos

* fixed headings

* fixed numerious typos

* Apply suggestions from code review

looks good now !!

Co-authored-by: Christopher Moyer <35463610+chri2547@users.noreply.github.com>

* Apply suggestions from code review

Thanks for the changes. It looks much better now

Co-authored-by: Christopher Moyer <35463610+chri2547@users.noreply.github.com>

* fixed the suggested changes and fixed minor typos

* Apply suggestions from code review

thanks for the improvements. looks polished now!!

Co-authored-by: Christopher Moyer <35463610+chri2547@users.noreply.github.com>

* fixed download link

* fixed typo

* final adjustments

* corrects spelling

* makes prettier

---------

Co-authored-by: Christopher Moyer <35463610+chri2547@users.noreply.github.com>
Co-authored-by: Chris Moyer <chris.moyer@grafana.com>

NWRichmond committed:
Prometheus: avoid unnecessary network requests (#83342)

* perf: avoid unnecessary network requests

* test: restore mocks to undo `jest.replaceProperty`

* chore: mirror updates for Prometheus library

stephaniehingtgen committed:
Chore: fix shellchecks (#83421)

renovate[bot] committed:
Update dependency @types/node to v20.11.20

kaydelaney committed:
Scenes/VizPanel: Add support for panel repeat options (#81818)

Takashiidobe committed:
fix typos (#83414)

renovate[bot] committed:
Update dependency @types/react to v18.2.59

mgyongyosi committed:
Auth: Revert provider list change (#83435)

* Load auth.grafananet as the last provider

* skip test

fabrizio-grafana committed:
Fix linting of docs file (#83441)

iwysiu committed:
CloudWatch: Move SessionCache onto the instance (#83278)

renovate[bot] committed:
Update dependency @types/react-color to v3.0.12

toddtreece committed:
Chore: Point to xorm fork in go.mod (#83436)

iwysiu committed:
CloudWatch: Fetch externalId from settings instead of env (#83332)

iwysiu committed:
CloudWatch: Refactor "getDimensionValuesForWildcards" (#83335)

yuri-tceretian committed:
Alerting: Deprecate max_annotations_to_keep and max_annotation_age in [alerting] configuration section (#83266)

* introduce new config section [unified_alerting.state_history.annotations] and deprecate settings in [alerting]

Co-authored-by: brendamuir <100768211+brendamuir@users.noreply.github.com>

renovate[bot] committed:
Update dependency @types/react-window-infinite-loader to v1.0.9 (#83445)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
Update dependency @types/semver to v7.5.8

renovate[bot] committed:
Update dependency dompurify to v3.0.9

renovate[bot] committed:
Update dependency nanoid to v5.0.6

leeoniya committed:
Graph (old): Migrate right y axis to TimeSeries via custom.axisPlacement (#83452)

adela-almasan committed:
Legend: Allow item copy in Table mode (#83319)

leeoniya committed:
BarChart: TooltipPlugin2 (#80920)

Co-authored-by: Adela Almasan <adela.almasan@grafana.com>

leeoniya committed:
TimeSeries: Don't re-init chart with bars style on data updates (#83355)

adela-almasan committed:
Histogram: Replace null values (#83195)

chalapat committed:
Update docs/sources/setup-grafana/configure-grafana/_index.md

Co-authored-by: Christopher Moyer <35463610+chri2547@users.noreply.github.com>

chalapat committed:
Update docs/sources/setup-grafana/configure-grafana/_index.md

Co-authored-by: Christopher Moyer <35463610+chri2547@users.noreply.github.com>

chalapat committed:
Update docs/sources/setup-grafana/configure-grafana/_index.md

Co-authored-by: Christopher Moyer <35463610+chri2547@users.noreply.github.com>

chalapat committed:
Update docs/sources/setup-grafana/configure-grafana/_index.md

Co-authored-by: Christopher Moyer <35463610+chri2547@users.noreply.github.com>

chalapat commented:
hi all, Something wrong happened to my push. it is showing too many files. i will close this and contribute via a different pull request. Sorry for the inconvenience.
